### ISSUE TITLE
7$: Better error if generator `*` is before `function` 

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#267]
In the following code, the user tried to declare a generator function:
```javascript
*function myGenerator() {
  yield 'hello';
}
```
quick-lint-js reports confusing errors for this code. quick-lint-js should instead report something like the following:
> generator function '*' belongs after the 'function' keyword

**Comment by Akhil132-dev** [comment]
I think it would be right =>  function* myGenerator() {
  yield 'hello';
}

**Comment by singalhimanshu** [comment]
What should be the error recovery in this case? Should we handle it as if it is a generator function?

**Comment by strager** [comment]
> What should be the error recovery in this case? Should we handle it as if it is a generator function?

Yes, handle it as a generator function.

**Comment by singalhimanshu** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by singalhimanshu** [comment]
What should be the error for this case - 
```js
*
function f(x) { yield x; }
```
Should it be the same error i.e. `error_generator_function_star_belongs_before_name` or something else?

**Comment by strager** [comment]
> What should be the error for this case -
> 
> ```js
> *
> function f(x) { yield x; }
> ```

Good question.

I see three reasonable options:

1. unexpected `*`
2. missing operand for `*` (this is what is reported today)
3. generator function star belongs after `function` keyword (before function name)

I don't really care which we do. I have a slight preference for 1 or 2.
---

Unfortunately, the following code is legal, so we need to be a bit careful with when we issue a diagnostic:
```js
console.log('hi')
*function f() {}
console.log('yup')

f() // runtime error
```
